src/mon/NVMeofGwSerialize.h

```diff
@@ -308,23 +313,35 @@ inline  void decode(
 }
 
 inline void encode(const NvmeGwClientState& state,  ceph::bufferlist &bl, uint64_t features) {
-  ENCODE_START(1, 1, bl);
+  uint8_t version = 1;
+  if (HAVE_FEATURE(features, NVMEOF_BEACON_DIFF)) {
+     version = 2;
+  }
+  ENCODE_START(version, version, bl);
   encode(state.group_id, bl);
   encode(state.gw_map_epoch, bl);
   encode (state.subsystems, bl, features);
   encode((uint32_t)state.availability, bl);
+  if (version >= 2) {
+    encode((uint64_t)state.last_beacon_seq_number, bl);
+    encode(state.last_beacon_seq_ooo, bl);
+  }
   ENCODE_FINISH(bl);
 }
 
 inline  void decode(
   NvmeGwClientState& state,  ceph::bufferlist::const_iterator& bl) {
-  DECODE_START(1, bl);
+  DECODE_START(2, bl);
   decode(state.group_id, bl);
   decode(state.gw_map_epoch, bl);
   decode(state.subsystems, bl);
   uint32_t avail;
   decode(avail, bl);
   state.availability = (gw_availability_t)avail;
+  if (struct_v >= 2) {
```

Comment:
This check is both correct and necessary, regardless of whether encode includes these fields conditionally.